runtime/src/main/java/io/quarkiverse/githubapp/runtime/github/GitHubService.java

```diff
@@ -222,41 +207,35 @@ private GitHub createApplicationGitHub() {
     private static class JwtTokenProvider {
         private final String appId;
         private final PrivateKey privateKey;
-        private final Set<String> missingPropertyKeys = new HashSet<>();
         private final JwtTokenCreator jwtTokenCreator;
+        private final ConfigChecker configChecker;
+        private final LaunchMode launchMode;
 
-        public JwtTokenProvider(GitHubAppRuntimeConfig gitHubAppRuntimeConfig, JwtTokenCreator jwtTokenCreator) {
+        public JwtTokenProvider(GitHubAppRuntimeConfig gitHubAppRuntimeConfig, JwtTokenCreator jwtTokenCreator,
+                ConfigChecker configChecker, LaunchMode launchMode) {
             this.jwtTokenCreator = jwtTokenCreator;
             if (gitHubAppRuntimeConfig.appId.isPresent()) {
                 appId = gitHubAppRuntimeConfig.appId.get();
             } else {
                 appId = null;
-                missingPropertyKeys.add("quarkus.github-app.app-id");
```

Comment:
It's a shame that we're now doing configuration fetching and validation in completely separate classes...

Can't you move the responsibility of accessing configuration to `ConfigChecker` (probably to be renamed `CheckedConfigProvider` or something)?

Then in `GitHubService you'd just call `checkedConfigProvider.appId()` and that would trigger checking as necessary.